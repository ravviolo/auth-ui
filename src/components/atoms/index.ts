export { Button } from './Button/Button';
export { Checkbox } from './Checkbox/Checkbox';
export { Divider } from './Divider/Divider';
export { Heading } from './Heading/Heading';
export { IconButton } from './IconButton/IconButton';
export { InputField } from './InputField/InputField';
export { InputLabel } from './InputLabel/InputLabel';
export { TextButton } from './TextButton/TextButton';
