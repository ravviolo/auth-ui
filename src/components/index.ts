export * from './templates/LoginTemplate/LoginTemplate';
export * from './templates/SignUpTemplate/SignUpTemplate';
