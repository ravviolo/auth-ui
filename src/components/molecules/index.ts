export { AuthFooter } from './AuthFooter/AuthFooter';
export { AuthHeader } from './AuthHeader/AuthHeader';
export { AuthSocials } from './AuthSocials/AuthSocials';
export { EmailInput } from './EmailInput/EmailInput';
export { PasswordInput } from './PasswordInput/PasswordInput';
