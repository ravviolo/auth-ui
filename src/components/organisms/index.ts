export * from './LoginForm/LoginForm';
export * from './SignUpForm/SignUpForm';
