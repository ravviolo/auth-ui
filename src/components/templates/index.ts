export * from './LoginTemplate/LoginTemplate';
export * from './SignUpTemplate/SignUpTemplate';
