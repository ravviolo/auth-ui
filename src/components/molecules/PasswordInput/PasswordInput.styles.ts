import styled from 'styled-components';

import { Flex } from 'components/atoms';

export const InputContainer = styled(Flex)`
  gap: 0.2em;
`;
