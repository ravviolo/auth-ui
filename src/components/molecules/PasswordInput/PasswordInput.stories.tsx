import { ComponentMeta, ComponentStory } from '@storybook/react';

import { PasswordInput } from 'components/molecules';

export default {
  title: 'Molecules/PasswordInput',
  component: PasswordInput,
  args: {
    name: 'password',
    id: 'password',
  },
} as ComponentMeta<typeof PasswordInput>;

export const Default: ComponentStory<typeof PasswordInput> = (args) => (
  <PasswordInput {...args} />
);
Default.storyName = 'PasswordInput';
