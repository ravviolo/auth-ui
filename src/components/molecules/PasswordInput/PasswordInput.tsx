import React from 'react';

import { InputField, InputLabel } from 'components/atoms';

import * as S from './PasswordInput.styles';

interface Props {
  id: string;
  name: string;
}

export const PasswordInput = React.forwardRef<HTMLInputElement, Props>(
  ({ id, name }, ref) => {
    return (
      <S.InputContainer direction="column" testId="flex-password-input">
        <InputLabel htmlFor={id} label="Password" />
        <InputField
          ref={ref}
          id={id}
          name={name}
          testId="input-field-password"
          type="password"
        />
      </S.InputContainer>
    );
  }
);

PasswordInput.displayName = 'PasswordInput';
