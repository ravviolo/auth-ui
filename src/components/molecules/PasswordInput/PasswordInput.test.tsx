import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';

import { PasswordInput } from 'components/molecules';

import { Props as PasswordInputProps } from './PasswordInput';

describe('PasswordInput', () => {
  const props: PasswordInputProps = {
    id: 'test-password',
    name: 'test-password',
    testId: 'password-input-test-id',
  };

  it('should render input of type password', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputField = within(passwordInput).getByTestId(
      'input-field-password-test-id'
    );

    expect(passwordInputField).toHaveAttribute('type', 'password');
  });

  it('should correctly assign id, name attributes received via props', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputField = within(passwordInput).getByTestId(
      'input-field-password-test-id'
    );

    expect(passwordInputField).toHaveAttribute('id', 'test-password');
    expect(passwordInputField).toHaveAttribute('name', 'test-password');
  });

  it('should semantically connect input label with input field', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputLabel = within(passwordInput).getByTestId(
      'input-label-password-test-id'
    );
    const passwordInputField = within(passwordInput).getByTestId(
      'input-field-password-test-id'
    );

    expect(passwordInputLabel).toHaveAttribute(
      'for',
      passwordInputField.getAttribute('id')
    );
  });

  it('should render initially empty input', () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputField = within(passwordInput).getByTestId(
      'input-field-password-test-id'
    );

    expect(passwordInputField).toHaveValue('');
  });

  it("should display 'Password' as input label", () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');
    const passwordInputLabel = within(passwordInput).getByTestId(
      'input-label-password-test-id'
    );

    expect(passwordInputLabel).toHaveTextContent('Password');
  });

  it('should display entered text in input field', async () => {
    render(<PasswordInput {...props} />);

    const passwordInput = screen.getByTestId('password-input-test-id');

    const passwordInputField = within(passwordInput).getByTestId(
      'input-field-password-test-id'
    );

    await userEvent.type(passwordInputField, 'Test password');

    expect(passwordInputField).toHaveValue('Test password');
  });

  it('should match snapshot', () => {
    const tree = create(<PasswordInput {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
