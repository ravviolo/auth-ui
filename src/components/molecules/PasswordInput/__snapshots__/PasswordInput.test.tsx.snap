// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput matches snapshot 1`] = `
<div
  className="sc-jSMfEi bQrDqy sc-ksZaOG cRa-dNN"
  data-testid="test-password-input"
  direction="column"
>
  <label
    className="sc-papXJ daxRFt"
    data-testid="input-label-password"
    htmlFor="test-password"
  >
    Password
  </label>
  <input
    className="sc-ftvSup dLJLzu"
    data-testid="input-field-password"
    id="test-password"
    name="test-password"
    type="password"
  />
</div>
`;
